contrib/opencensus-ext-datadog/opencensus/ext/datadog/traces.py

```diff
@@ -0,0 +1,374 @@
+                meta["opencensus.status_description"] = span.get("status").get(
+                    "message")
+
+            atts = span.get("attributes").get("attributeMap")
+            atts_to_metadata(atts, meta=meta)
+
+            dd_span["meta"] = meta
+            dd_trace.append(dd_span)
+
+        return dd_trace
+
+
+def atts_to_metadata(atts, meta={}):
+    """Translate the attributes to Datadog meta format.
+
+    :type atts: dict
+    :param atts: Attributes dictionary
+
+    :rtype: dict
+    :returns: meta dictionary
+    """
+    for key, elem in atts.items():
+        value = value_from_atts_elem(elem)
+        if value != "":
+            meta[key] = value
+
+    return meta
+
+
+def value_from_atts_elem(elem):
+    """ value_from_atts_elem takes an attribute element and retuns a string value
+
+    :type elem: dict
+    :param elem: Element from the attributes map
+
+    :rtype: str
+    :return: A string rep of the element value
+    """
+    if elem.get('string_value') is not None:
+        return elem.get('string_value').get('value')
+    elif elem.get('int_value') is not None:
+        return str(elem.get('int_value'))
+    elif elem.get('bool_value') is not None:
+        return str(elem.get('bool_value'))
+    elif elem.get('double_value') is not None:
+        return str(elem.get('double_value').get('value'))
+    return ""
+
+
+def to_dd_type(oc_kind):
+    """ to_dd_type takes an OC kind int ID and returns a dd string of the span type
+
+    :type oc_kind: int
+    :param oc_kind: OC kind id
+
+    :rtype: string
+    :returns: A string of the Span type.
+    """
+    if oc_kind == 2:
+        return "client"
+    elif oc_kind == 1:
+        return "server"
+    else:
+        return "unspecified"
+
+
+def new_trace_exporter(option):
+    """ new_trace_exporter returns an exporter
+    that exports traces to Datadog.
+    """
+    if option.service == "":
+        raise ValueError("Service can not be empty string.")
+
+    exporter = DatadogTraceExporter(options=option)
+    return exporter
+
+
+def convert_id(str_id):
+    """ convert_id takes a string and converts that to an int that is no
+    more than 64 bits wide. It does this by first converting the string
+    to a bit array then taking up to the 64th bit and creating and int.
+    This is equivlent to the go-exporter ID converter
+    ` binary.BigEndian.Uint64(s.SpanContext.SpanID[:])`
+
+    :type str_id: str
+    :param str_id: string id
+
+    :rtype: int
+    :returns: An int that is no more than 64 bits wide
+    """
+    id_bitarray = bitarray.bitarray(endian='big')
+    id_bitarray.frombytes(str_id.encode())
+    cut_off = len(id_bitarray)
+    if cut_off > 64:
+        cut_off = 64
+    id_cutoff_bytearray = id_bitarray[:cut_off].tobytes()
+    id_int = int(codecs.encode(id_cutoff_bytearray, 'hex'), 16)
+    return id_int
+
+
+# 2019-09-19T14:05:15.808570Z
+TIME_FMT = "%Y-%m-%dT%H:%M:%S.%fZ"
```

Comment:
Potentially we can reuse the logic: https://github.com/census-instrumentation/opencensus-python/blob/a6d6834974b4ebf273905e2d903f3f1c39cc8e4e/opencensus/common/utils/__init__.py#L29